dartclient/core.py

```diff
@@ -408,9 +447,13 @@ def sync_action(self, action_name, workflow, callback, dataset=None, subscriptio
         if action:
             action = callback(action)
             if dataset:
-                action.args.dataset_id = dataset.id
+                action.data.args = {
```

Comment:
You are going to override the existing data.args dictionary as set by callback if you assign a dictionary here to action.data.args. This should probably be something like:
```
if not action.data.args:
    action.data.args = {}
action.data.args['dataset_id'] = dataset.id
```